src/utils/appDataClient.ts

```diff
@@ -51,17 +136,43 @@ export const hotspotOnChain = async (address: string) => {
 }
 
 export const getChainVars = async (keys?: string[]) => {
-  return heliumHttpClient.vars.get(keys)
+  try {
+    return privateHeliumClient.vars.get(keys)
+  } catch (error) {
+    try {
+      return heliumHttpClient.vars.get(keys)
+    } catch (err) {
+      throw new Error(
+        `Helium API call is failed while getting the chain vars: ${err}`,
+      )
+    }
+  }
 }
 
 export const getHotspotsLastChallengeActivity = async (
   gatewayAddress: string,
 ) => {
-  const hotspotActivityList = await heliumHttpClient
-    .hotspot(gatewayAddress)
-    .activity.list({
-      filterTypes: ['poc_receipts_v1', 'poc_request_v1'],
-    })
+  let hotspotActivityList: ResourceList<AnyTransaction>
+  try {
+    hotspotActivityList = await privateHeliumClient
+      .hotspot(gatewayAddress)
+      .activity.list({
+        filterTypes: ['poc_receipts_v1', 'poc_request_v1'],
+      })
+  } catch (error) {
+    try {
+      hotspotActivityList = await heliumHttpClient
+        .hotspot(gatewayAddress)
+        .activity.list({
+          filterTypes: ['poc_receipts_v1', 'poc_request_v1'],
+        })
+    } catch (err) {
+      throw new Error(
+        `Helium API call is failed while getting the hotspot activity list: ${err}`,
```

Comment:
Helium API request failed while trying to retrieve hotspot {gatewayAddress} activity.